### ISSUE TITLE
Issue in dataloader -- total perplexity in paper incorrect

### ISSUE THREAD

**Issue by UsmannK** [shaoxiongji/fed-att#4]
Hi, your dataloader uses the default collate_fn in pytorch 0.4.1. This truncates every batch to the length of the shortest sentence in that batch. By extension your experiments are only operating on about 20% of the total datasets. 

When fixed, this increases perplexity to about 300 for both FedAtt and FedAvg on the Penn Treebank dataset.

**Comment by UsmannK** [comment]
Can you provide the parameters used to reproduct the results in the paper? epochs, local epochs, learning rate, etc
